queue/server.go

```diff
@@ -0,0 +1,121 @@
+package queue
+
+import (
+	"sync"
+
+	"context"
+	"errors"
+	"time"
+
+	"github.com/centrifuge/gocelery"
+	logging "github.com/ipfs/go-log"
+)
+
+const (
+	BlockHeightParam string = "BlockHeight"
+	TimeoutParam     string = "Timeout"
+)
+
+var log = logging.Logger("queue-server")
+
+// Config is an interface for queue specific configurations
+type Config interface {
+
+	// GetNumWorkers gets the number of background workers to initiate
+	GetNumWorkers() int
+
+	// GetWorkerWaitTime gets the worker wait time for a task to be available while polling
+	// increasing this may slow down task execution while reducing it may consume a lot of CPU cycles
+	GetWorkerWaitTimeMS() int
+}
+
+// TaskType is a task to be queued in the centrifuge node to be completed asynchronously
+type TaskType interface {
```

Comment:
I'm not sure if the `TaskType` is the correct name for this since it only has one method that returns name of the task. 

I noticed we are not calling the interface method anywhere. Do we really need this ?
I would rather have us use `CelereyTask` interface which every task implements.